src/lib.rs

```diff
@@ -333,6 +415,30 @@ fn set(
     Ok(())
 }
 
+/// The only entry for define a fail point.
+///
+/// When a fail point is defined, it's reference via the full module path and name in the
+/// format of `crate::module::submodule::fail_point_name`. For example, library A defines
+/// a fail point in lib.rs as follow:
+///
+/// ```rust
+/// pub fn f() {
+///     fail_point!("p1");
+/// }
+///
+/// mod M {
+///     fn f() {
+///         fail_point!("p2");
+///     }
+/// }
+/// ```
+/// The full name of fail point `p1` is `A::p1`, and for `p2` is `A::M::p2`.
+///
+/// `$e` is used to transform a string to the return type of outer function or closure. If
+/// the return type is `()`, then you can use shortcut `fail_point!($name)`.
```

Comment:
return -> returned
shortcut `fail_point!($name)` -> the `fail_point!($name)` shortcut.